Replace all occurrences of Person with Patient

w-yeehong opened the pull request:
Since we are handling `Patient`, we should replace all references to `Person` with `Patient`.

Would need to check whether I missed out anything.

w-yeehong committed:
Replace all occurrences of Person with Patient

LeeMingDe commented on src/test/data/JsonAddressBookStorageTest/invalidAndValidPatientAddressBook.json:
```diff
@@ -1,11 +1,11 @@
+  "patients": [ {
+    "name": "Valid Patient",
     "phone": "9482424",
     "email": "hans@example.com",
```
hmm, i thought that this field not supposed to be here anymore?

w-yeehong replied:
Agreed! Let's remove the additional fields.

LeeMingDe reviewed (approved):
LGTM

raymondge reviewed (approved):
Nice Work! It's absolutely more covigent now!

w-yeehong committed:
Transfer SearchPatient*.java to Patient package

w-yeehong committed:
Package commands, parsers, and unit tests

- Transfer all Patient-related classes into their packages.
- Transfer all Room-related classes into their packages.

w-yeehong committed:
Update JSON data to reflect valid and invalid Patient

w-yeehong commented:
While not ideal, I decided to force push to my own repo in order to squash the commits that contained only minor changes for CheckStyle compliance.

I will be more careful when pushing to my repo in the future.
